src/rsa.rs

```diff
@@ -119,6 +119,27 @@ fn verify(padding: &RSAPadding, msg: untrusted::Input,
 }
 
 
+pub mod primitive {
+    use untrusted;
+    /// Lower-level verification function for RSA signatures.
+    ///
+    /// If you do not have the public key available as DER-encoded
+    /// PKCS#1-formatted ASN.1, this function can be used to pass in the raw
+    /// bytes for the public key components as `untrusted::Input` arguments.
+    ///
+    /// Takes an `RSAPadding` static to determine the padding algorithm used,
+    /// a message, a signature (whose length should match the length of public
+    /// exponent, `n`) and a tuple containing the parts of the public key (in
+    /// big-endian byte order). The key length must be between 2048 and 8192 bits.
+    pub fn verify(padding: &super::RSAPadding, msg: untrusted::Input,
+                  signature: untrusted::Input,
+                  (n, e): (untrusted::Input, untrusted::Input))
+                  -> Result<(), ()> {
+        super::verify(padding, msg, signature, (n, e), 2048)
```

Comment:
Please see my comments in the other PR where you renamed `RSA_PKCS1_2048_8192_SHA*_VERIFY` to `RSA_PKCS1_SHA*_2048_8192`. I had landed that change, and then I saw this change and decided to revert that change because this helped me see that we're going in the wrong direction.

As discussed previously, our goal is to eventually develop an API that treats ECDSA, RSA, etc. uniformly or at least as similarly as possible. In ECDSA, the only parameters you have are curve and digest algorithm. In RSA, you have min bits, max bits, digest algorithm, exponent, and padding scheme. In *ring*, I've chosen to use the following mapping:

| ECDSA                               | RSA                                  | EdDSA         |
|-------------------------------------|--------------------------------------|---------------|
| Algorithm = ECDSA | Algorithm = (RSA, Padding Scheme) | Algorithm = EdDSA |
| Curve                       | modulus min bits, modulus max bits, exponent min bits, exponent max bits)               | Curve |
| Digest Alg.                         | Digest Alg.                          | implicit      |
| Encoding (ASN.1, PKCS#11/WebCrypto) | implicit                             | implicit      |

Note that the exponent range is fixed at (2 bits, 33 bits). But, I'm thinking of changing that so that for the 3072-8192 variant (Suite B), at least, the minimum exponent length will be 16 bits.

So, anyway, I'm a little unsure how important it is to be explicit about the allowed exponent range, but at least this new primitive API should allow the caller to be explicit about the modulus range--at least the minimum modulus length.

Going back to your other PR, the issue you had was that `RSAPadding`s are named `RSA_<padding_scheme>_<digest_alg>` but RSA `SignatureAlgorithm`s are named similarly, but they put <min_modulus_bits>_<max_modulus_bits> in between. And, that affects this API because it takes `RSAPadding` as a parameter. But, maybe this API shouldn't take `RSAPadding`.

Ideally, this API should take `RSA_PKCS1_2048_8192_SHA256`, etc. as parameters instead, like `verify()` does, but with a type that isn't `VerificationAlgorithm` but instead something like `RSAParameters` or something. After all, this API is exactly like `verify` except that it takes its public key argument pre-parsed.

But, I'm not sure how to do that without making `VerificationAlgorithm` a trait, which I don't want to do for the reasons I explained before. If I have to choose between the main `verify` API forcing people to `use` traits or the `primitive` API forcing people to `use` traits, I'd prefer to put that inconvenience on the `primitive` API.

Maybe it is possible to change the type of `RSA_PKCS1_2048_8192_SHA256` et al. from `&'static VerificationAlgorithm` to `&'static <VerificationAlgorithm + RSAParameters>` and make `RSAParameters` a trait? I've never tried to do that before so i'm not sure.

Or, if that's all too complicated, then we can punt on trying to create a really uniform API and just, at least, make the minimum key size a parameter of this new function. The main thing I want to avoid doing is making the key size implicit like it is in this patch, because one of the most important design goals for *ring* is to avoid such implicit security-critical parameters.